base/bitarray.jl

```diff
@@ -797,6 +797,7 @@ function sizehint!(B::BitVector, sz::Integer)
 end
 
 function resize!(B::BitVector, n::Integer)
+    n = Int(n)
```

Comment:
I'm pretty sure in other places we do this kind of thing as
```julia
resize!(B::BitVector, n::Integer) = resize!(B,  Int(n))
function resize!(B::BitVector, n::Int)
   ...
```
The rationale being that this way, only the trivial `resize!(::BitVector, ::Integer)` needs to be specialized on different types (unless it is inlined away anyway), while only one specialization is needed for the actual implementation. Is that not a concern here? Or has the general landscape shifted such that that paradigm is obsolete?

Reply:
Tough call. There are a few advantages in keeping the number of methods small, as Julia's compiler does "world-splitting" (like union-splitting except making decisions based on existing methods rather than an explicit `Union`) when there are 3 or fewer methods.

We could add a `@nospecialize n` but I'm not sure I like that either.

Reply:
If the number of methods is an issue then
```julia
resize!(B::BitVector, n::Integer) = _resize!(B,  Int(n))
function _resize!(B::BitVector, n::Int)
   ...
```
would be the obvious variation to keep that in check.